### PR TITLE
API Extractor now validates release tags and autogenerates constructor summaries

### PR DESCRIPTION
**Pull request by pgonzal**
API Extractor now has two new features:
1. If a top-level package export is missing a release tag (`@alpha`, `@beta`, `@public`, `@internal`), this is reported as an error.
2. If a constructor is missing its summary, a default summary is generated: "Constructs a new instance of the ____ class"

Both of these features require the ability to examine parent nodes in the ApiItem tree, also introduced by this PR.

### CODE REVIEW

File: common/changes/@microsoft/gulp-core-build/pgonzal-ae-new-validation_2017-06-20-19-08.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/gulp-core-build",
+      "comment": "Added missing release tags for API Extractor",
```
Comment (qz2017) [inline common/changes/@microsoft/gulp-core-build/pgonzal-ae-new-validation_2017-06-20-19-08.json]: >Added [](start = 18, length = 5)

Add

File: libraries/api-extractor/src/ApiDefinitionReference.ts [hunk 2]
```diff
@@ -101,10 +101,10 @@ export default class ApiDefinitionReference {
    * and deconstructs it into an IApiDefinitionReference interface object.
```
Comment (chackman) [inline libraries/api-extractor/src/ApiDefinitionReference.ts]: Also document the behavior when an error is encountered - return value is undefined, error is logged via reportError delegate.

**Comment by qz2017** [comment]
:shipit:

**Comment by chackman** [comment]
:shipit:

**Pull request merged**